### PR TITLE
Clarifications

### PR DESCRIPTION
**Pull request by Susurrus**
Will need someone to confirm these changes, but getting responses seems to go fastest when I submit a PR instead of an issue, so here it is. I broke them up into individual commits so hopefully the non-contentious commits can get merged quickly. 1 & 4 might be wrong, 2 & 3 should be straight forward, and 5 might be contentious, so let's see what we get!

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Thanks! 1-4 are fine, 5 is not because that's not how this message is being used.

Its not a surprise a PR gets a quick response. If you want people to take an issue and fix it for them, you either have to convince them that they need this fix in general (which is a high bar for people being constantly overloaded) or pay other people who provide additional development bandwidth. If you do neither, there is no reason to address it.

**Comment by Susurrus** [comment]
Thanks, Lorenz, for checking this over so quickly. To merge should I remove commit 5 or do you want to cherry-pick 1-4?

**Comment by Susurrus** [comment]
Nevermind, I trimmed off the last commit (need to practice my Git skills anyways) and once Travis signs off, we should be good to go.

Though the one thing this merge doesn't address is WHEN the SETPOINT messages are active. You mentioned in #473 that there's a `OFFBOARD_CONTROL` mode that is pending to be included in MAVLink. Can we merge that in, even if it's tentative, as these messages are incomplete without those changes. I'd like to marge them as in beta or something because of this. Please advise.

**Comment by Susurrus** [comment]
Threw in a fix for #471 while I was at it since it seemed to fit with the theme of this PR.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1671,7 +1671,7 @@
                   <description>Autopilot supports the new param union message type.</description>
               </entry>
               <entry value="32" name="MAV_PROTOCOL_CAPABILITY_FTP">
-                  <description>Autopilot supports the new param union message type.</description>
+                  <description>Autopilot supports the new RESOURCE_REQUEST message type.</description>
```
Comment (vooon) [inline message_definitions/v1.0/common.xml]: Nope, it is about [`FILE_TRANSFER_PROTOCOL` (110)](http://mavlink.org/messages/common#FILE_TRANSFER_PROTOCOL) message.

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -1722,10 +1722,10 @@
                   <description>Not specified.</description>
               </entry>
               <entry value="1" name="MAV_BATTERY_TYPE_LIPO">
-                  <description>Lithium polymere battery</description>
+                  <description>Lithium polymer battery</description>
               </entry>
               <entry value="2" name="MAV_BATTERY_TYPE_LIFE">
-                  <description>Lithium ferrite battery</description>
+                  <description>Lithium iron battery</description>
```
Comment (vooon) [inline message_definitions/v1.0/common.xml]: Not sure. Chemistry called LiFePO4 (lithium ferrophosphate).

Comment (Susurrus) [inline message_definitions/v1.0/common.xml]: Should I change it to LiFePO4? I don't really care, but I've never heard someone refer to it as lithium ferrite, so I don't know if that's a good description. Maybe LiFePO4 is better?

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (Susurrus) [inline message_definitions/v1.0/common.xml]: Noted, I'll update that.

File: message_definitions/v1.0/common.xml (hunk 2 above)
Comment (vooon) [inline message_definitions/v1.0/common.xml]: Maybe. Wikipedia: [Lithium Iron Phosphate / LFP](https://en.wikipedia.org/wiki/Lithium_iron_phosphate_battery).

**Comment by Susurrus** [comment]
Found another issue and cleaned up both things pointed out by @vooon. Rebased to keep things looking nice.

**Comment by Susurrus** [comment]
@LorenzMeier @tridge can you guys give this one last look over? I think we're ready to merge at this stage.

**Pull request merged**